libs/server/Resp/Objects/ListCommands.cs

```diff
@@ -191,6 +192,60 @@ private unsafe bool ListPop<TGarnetApi>(int count, byte* ptr, ListOperation lop,
             return true;
         }
 
+        private bool ListBlockingPop<TGarnetApi>(int count, byte* ptr, ListOperation lop, ref TGarnetApi storageApi)
+            where TGarnetApi : IGarnetApi
+        {
+            if (count < 2)
+            {
+                return AbortWithWrongNumberOfArguments(lop.ToString(), count);
+            }
+
+            var keys = new ArgSlice[count - 1];
+
+            for (var i = 0; i < keys.Length; i++)
+            {
+                keys[i] = default;
+                if (!RespReadUtils.ReadPtrWithLengthHeader(ref keys[i].ptr, ref keys[i].length, ref ptr, recvBufferPtr + bytesRead))
+                    return false;
+            }
+
+            if (NetworkKeyArraySlotVerify(ref keys, false))
+            {
+                var bufSpan = new ReadOnlySpan<byte>(recvBufferPtr, bytesRead);
+                if (!DrainCommands(bufSpan, count))
+                    return false;
+                return true;
+            }
+
+            if (!RespReadUtils.ReadDoubleWithLengthHeader(out var timeout, out var parsed, ref ptr,
+                recvBufferPtr + bytesRead) || !parsed)
+                return false;
+
+            var arrKeys = keys.Select(k => k.Span.ToArray()).ToArray();
+            var result = itemBroker.GetListItemAsync(arrKeys, lop, this, timeout).Result;
```

Comment:
Wait on a SemaphoreSlim here, that can be canceled (using a cancellation token) if the socket is killed or the session dies for some reason (i.e., session Dispose gets called).